### PR TITLE
update login

### PR DESCRIPTION
**Pull request by WuJun2016**
1. Add login logic using cypresa.command.add
1. Fix a bug in the use of cy.visit  （We recommend setting a baseUrl when using cy.visit()）

### CODE REVIEW

**Comment by n313893254** [comment]
I've pushed a minor tidy up, @WuJun2016 Could you take a look?

File: cypress/pageobjects/virtualmachine.po.ts [hunk 1]
```diff
@@ -27,16 +26,16 @@ interface ValueInterface {
 export class VmsPage {
   public goToList() {
     cy.visit(constants.vmPage);
-    cy.intercept('GET', '/v1/harvester/kubevirt.io.virtualmachines').as('loadVM');
-    cy.wait('@loadVM')
-    expect(cy.url().should('eq', constants.vmPage));
+
+    cy.get('nav.side-nav .list-unstyled').find('li').contains('Virtual Machines').click();
+    cy.get('main .outlet header h1').then($el => {
```
Comment (noahgildersleeve) [inline cypress/pageobjects/virtualmachine.po.ts:31]: These gets should be pointed at a class member that is assigned the css selectors. Such as
```
private sideBar = 'nav.side-nav .list-unstyled';
cy.get(this.sideBar).find
```

File: cypress/support/commands.js [hunk 2]
```diff
@@ -23,3 +21,26 @@
...
+    cy.wait('@authProviders').then(res => {
+      const { CSRF } = cookie.parse(document.cookie);
+      cy.request({
+        method: 'POST',
+        url: '/v3-public/localProviders/local?action=login',
+        body: {
+          "description":"UI session",
+          "responseType":"cookie",
+          "username":"admin",
+          "password":"password1234"
```
Comment (noahgildersleeve) [inline cypress/support/commands.js:38]: Should this be pointing at the env username/passwords instead of being static?

**Review by noahgildersleeve — CHANGES_REQUESTED** [review]

File: cypress/support/commands.js (hunk 2 above)
Comment (WuJun2016) [reply cypress/support/commands.js:38]: Sorry, I forgot to change it back when I tested it.

File: cypress/pageobjects/image.po.ts [hunk 3]
```diff
@@ -35,11 +35,10 @@ export class ImagePage {
   }
 
   public goToList() {
-    // cy.get('.nav').contains('Images').click();
     cy.visit(constants.imagePage);
-    cy.intercept('GET', '/v1/harvester/harvesterhci.io.virtualmachineimages').as('goToImageList');
-    cy.wait('@goToImageList');
-    cy.url().should('eq', constants.imagePage);
+    cy.get('main .outlet header h1').then($el => {
```
[the remark this answers is not in the document]
Comment (WuJun2016) [reply cypress/pageobjects/image.po.ts]: Done

File: cypress/pageobjects/virtualmachine.po.ts (hunk 1 above)
Comment (WuJun2016) [reply cypress/pageobjects/virtualmachine.po.ts]: Done

File: cypress/pageobjects/virtualmachine.po.ts [hunk 4]
```diff
@@ -27,16 +26,16 @@ interface ValueInterface {
...
+      cy.url().should('exist', constants.vmPage);
+    })
   }
 
   public goToCreate() {
     this.goToList()
-    cy.intercept('GET', '/v1/harvester/configmaps').as('loadVMCreate');
-    cy.contains('Create').click()
-    cy.wait('@loadVMCreate')
+    cy.get('.outlet .actions-container').find('a').contains(' Create ').click();
```
[the remark this answers is not in the document]
Comment (WuJun2016) [reply cypress/pageobjects/virtualmachine.po.ts]: Done

File: cypress/support/commands.js (hunk 2 above)
Comment (WuJun2016) [reply cypress/support/commands.js]: Done

**Comment by WuJun2016** [comment]
@noahgildersleeve   Done,  PTAL, thanks a lot.

File: cypress/pageobjects/virtualmachine.po.ts [hunk 5]
```diff
@@ -140,13 +143,18 @@ export class VmsPage {
...
   public init() {
-    cy.intercept('GET', `/v1/harvester/harvesterhci.io.virtualmachineimages`).as('getImages');
-
     image.goToList();
-    
-    cy.wait('@getImages').then(res => {
-      expect(res.response?.statusCode).to.equal(200);
-      const images = res.response?.body?.data || []
+    cy.request({
+      url: 'v1/harvester/harvesterhci.io.virtualmachineimages',
```
Comment (noahgildersleeve) [inline cypress/pageobjects/virtualmachine.po.ts:148]: This should be saved as a constant in the constants file in case we need to change it easily later.

**Review by noahgildersleeve — CHANGES_REQUESTED** [review]
It looks good there's just one change for a constant I would suggest.

File: cypress/pageobjects/virtualmachine.po.ts (hunk 5 above)
Comment (WuJun2016) [reply cypress/pageobjects/virtualmachine.po.ts]: Done

**Pull request merged**